### PR TITLE
Don't set email if invalid in user:add command

### PR DESCRIPTION
**Pull request by tomneedham**
## Related Issue
https://github.com/owncloud/core/issues/28425

## Motivation and Context
Otherwise invalid emails can be set by accident 

## How Has This Been Tested?
OCC commands with invalid and valid email addreses

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by phil-davis — APPROVED** [review]
works
the question is now, should it work like this (create the user, but just not set any email address), or
not create the user at all

**Comment by phil-davis** [comment]
@PVince81 @DeepDiver1975 or whoever, what do you think should be the behavior when the email address is not valid?
Add the user anyway, without an email address, or give an error and do not add the user at all?

**Comment by PVince81** [comment]
I'd say don't add the user at all as the whole user entity failed validation by having a wrong email address

File: core/Command/User/Add.php [hunk 1]
```diff
@@ -136,8 +142,14 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 		}
 
 		if ($input->getOption('email')) {
-			$user->setEMailAddress($input->getOption('email'));
-			$output->writeln('Email address set to  "' . $user->getEMailAddress() . '"');
+			// Validate first
+			if(!$this->mailer->validateMailAddress($input->getOption('email'))) {
+				// Invalid! Error
+				$output->writeln('<error>Invalid email address supplied</error>');
```
Comment (PVince81) [inline core/Command/User/Add.php]: please throw / return error code

Comment (phil-davis) [inline core/Command/User/Add.php]: The validation will have to happen before:
```
$this->userManager->createUser()
```
and throw an error there, before the user is created.
Then the code to create the email address happen around here, after the user is created.

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
@tomneedham please backport to stable10

**Pull request merged**